src/toplevel.c

```diff
@@ -62,28 +62,49 @@ jl_module_t *jl_new_main_module(void)
     return old_main;
 }
 
+int jl_module_reqs_met(jl_expr_t *reqs, jl_module_t *newm)
+{
+    int reeval = !newm;
+    size_t i, tlen = jl_array_len(reqs->args);
+    for (i = 0; i < tlen; i++) {
+        jl_expr_t *req = (jl_expr_t*)jl_exprarg(reqs, i);
+        if (!jl_is_expr(req) || req->head != tuple_sym) {
+            jl_error("syntax: malformed module-requires statement");
+        }
+        size_t j, rlen = jl_array_len(req->args);
+        jl_module_t *m = jl_main_module;
+        for (j = 0; j < rlen; j++) {
```

Comment:
It looks like this should share code with `eval_import_path`. We might want to handle leading dots, and you also have to be very careful how bindings are checked.